### PR TITLE
plugins/oil: init + tests

### PR DESCRIPTION
**Pull request by GaetanLepage**
Add support for the [oil.nvim](https://github.com/stevearc/oil.nvim) plugin.

### CODE REVIEW

File: plugins/utils/oil.nix [hunk 1]
```diff
@@ -0,0 +1,433 @@
...
+with lib; let
+  cfg = config.plugins.oil;
+  helpers = import ../helpers.nix {inherit lib;};
+
+  mkSizeOption = default: desc:
+    helpers.defaultNullOpts.mkNullable
+    (with types;
+      oneOf [
+        int
+        float
```
Comment (traxys) [inline plugins/utils/oil.nix:16]: Instead of a float maybe a `numbers.between 0.0 1.0` ?

File: plugins/utils/oil.nix [hunk 2]
```diff
@@ -0,0 +1,433 @@
...
+    '';
+
+    minWidth = mkSizeOption "[40 0.4]" ''
+      Width dimensions can be integers or a float between 0 and 1 (e.g. 0.4 for 40%).
+      Can be a single value or a list of mixed integer/float types.
+      minWidth = [40 0.4] means "the greater of 40 columns or 40% of total".
+    '';
+
+    width =
+      helpers.mkNullOrOption (with types; either int float)
```
Comment (traxys) [inline plugins/utils/oil.nix:36]: same

File: plugins/utils/oil.nix [hunk 3]
```diff
@@ -0,0 +1,433 @@
...
+        helpers.defaultNullOpts.mkStr "trash-put"
+        "Change this to customize the command used when deleting to trash.";
+
+      promptSaveOnSelectNewEntry =
+        helpers.defaultNullOpts.mkBool true
+        "Selecting a new/moved/renamed file or directory will prompt you to save changes first.";
+
+      keymaps =
+        helpers.defaultNullOpts.mkNullable
+        (with types; attrs)
```
Comment (traxys) [inline plugins/utils/oil.nix:269]: types.attrs would be shorter

**Review by traxys — CHANGES_REQUESTED** [review]

File: plugins/utils/oil.nix [hunk 4]
```diff
@@ -0,0 +1,433 @@
...
+  cfg = config.plugins.oil;
+  helpers = import ../helpers.nix {inherit lib;};
+
+  mkSizeOption = default: desc:
+    helpers.defaultNullOpts.mkNullable
+    (with types;
+      oneOf [
+        int
+        (numbers.between 0.0 1.0)
+        (listOf (either int float))
```
Comment (traxys) [inline plugins/utils/oil.nix:17]: You missed this one

**Review by traxys — CHANGES_REQUESTED** [review]

File: plugins/utils/oil.nix (hunk 4 above)
Comment (GaetanLepage) [reply plugins/utils/oil.nix]: Nice catch

**Review by traxys — APPROVED** [review]

**Pull request merged**